File: mod28/main.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class CountNumbersWithSumGreaterThan10
{
    public static int CountNumbersWithSumGreaterThan10Method(List<int> numbers)
    {
        int count = 0;
        foreach (int number in numbers)
        {
            int sum = 0;
            int temp = number;
            while (temp > 0)
            {
                sum += temp % 10;
                temp /= 10;
            }
            if (sum > 10)
            {
                count++;
            }
        }
        return count;
    }

    public static void Main(string[] args)
    {
        Console.Write("Введите список чисел, разделенных пробелами: ");
        string input = Console.ReadLine();
        List<int> numbers = new List<int>();
        string[] inputNumbers = input.Split(' ');

        foreach (string numStr in inputNumbers)
        {
            if (int.TryParse(numStr, out int num))
            {
                numbers.Add(num);
            }
            else
            {
                Console.WriteLine($"Некорректный ввод: {numStr}. Это значение будет пропущено.");
            }
        }

        int result = CountNumbersWithSumGreaterThan10Method(numbers);
        Console.WriteLine($"Количество чисел с суммой цифр > 10: {result}");
    }
}

File: mod28/main.rb
def count_numbers_with_sum_greater_than_10(numbers)
    count = 0
    numbers.each do |number|
      sum = 0
      temp = number
      while temp > 0
        sum += temp % 10
        temp /= 10
      end
      count += 1 if sum > 10
    end
    count
  end
  
  print 'Введите список чисел, разделенных пробелами: '
  input = gets.chomp
  input_numbers = input.split(' ')
  
  numbers = []
  input_numbers.each do |num_str|
    num = num_str.to_i
    unless num.zero? && num_str != '0' # Проверка на некорректный ввод
      numbers << num
    else
      puts "Некорректный ввод: #{num_str}. Это значение будет пропущено."
    end
  end
  
  result = count_numbers_with_sum_greater_than_10(numbers)
  puts "Количество чисел с суммой цифр > 10: #{result}"